poppler 21.11.0 rebuild for cairo 1.16.0 pinning

cbouss opened the pull request:
Changes:
- Rebuild for cairo pinning to 1.16.0. 
- Remove s390x skip, as boost-cpp is now available for s390x.
- Updated metadata.

Dev: https://gitlab.freedesktop.org/poppler/poppler/-/tree/poppler-21.11.0
License: https://gitlab.freedesktop.org/poppler/poppler/-/blob/poppler-21.11.0/COPYING

Jira: https://anaconda.atlassian.net/browse/DSNC-4947

TODO: abs.yaml and conda_build_config.yaml to be removed before merging.

cbouss committed:
cairo rebuild

cbouss committed:
don't skip s390x

cbouss committed:
add ld64.so to missing_dso_whitelist

varlackc reviewed (commented):
I am wondering if we should use the python skip instead of directly stating the pinning for python.

https://github.com/AnacondaRecipes/poppler-feedstock/blob/cairo_rebuild/recipe/meta.yaml#L53

tobijk commented on abs.yaml:
```diff
@@ -0,0 +1,2 @@
+channels: 
+  cbouss: cairo
```
Don't forget to remove this :)

cbouss committed:
use python skip instead of pinning python

cbouss commented:
Addressed comment from Maxwell by new commit.

However, the build now fails because since the last time, aggregate cbc was updated and there is now a dependency on libxcb which creates problems.

cbouss committed:
add ppc64le workaround

cbouss committed:
remove temp files

cbouss commented:
Hmm now there is an intermittent issue when building poppler.
We sometimes get ModuleNotFoundError: No module named 'giscanner._giscanner' 

I cannot merge this as it is. I either have to disable introspection or find the root cause and correct it.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_738pet4nar/clone/recipe/meta.yaml:99: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_738pet4nar/clone/recipe/meta.yaml:99: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>
